.github/workflows: use latest stable cilium-cli release

tklauser opened the pull request:
Use the `stable.txt` file in the cilium-cli repo to always fetch the
latest stable release.

rolinh commented on .github/workflows/conformance-aks.yaml:
```diff
@@ -152,7 +152,7 @@ jobs:
         run: |
-          curl -LO https://github.com/cilium/cilium-cli/releases/download/v0.8.4/cilium-linux-amd64.tar.gz
+          curl -LO https://github.com/cilium/cilium-cli/releases/download/v0.8.5/cilium-linux-amd64.tar.gz
           sudo tar xzvfC cilium-linux-amd64.tar.gz /usr/bin
```
Not directly related to this PR but I have 2 questions/suggestions:

- Why is the checksum not being used to verify the tarball?
- If we added a `stable.txt` file to the cilium-cli repo similarly to what is done in the Hubble and Cilium repos, we wouldn't have to open PRs to update Cilium CLI every time a new release is tagged.

TLDR, I'm suggesting to do something like this:
```
export CILUM_CLI_VERSION=$(curl -s https://raw.githubusercontent.com/cilium/cilium-cli/master/stable.txt)
curl -L --remote-name-all https://github.com/cilium/cilium-cli/releases/download/$CILUM_CLI_VERSION/cilium-linux-amd64.tar.gz{,.sha256sum}
sha256sum --check cilium-linux-amd64.tar.gz.sha256sum
sudo tar xzvfC cilium-linux-amd64.tar.gz /usr/local/bin
rm cilium-linux-amd64.tar.gz{,.sha256sum}
```

tklauser replied:
Good points. Let's change it in this PR already. I'll send a PR adding `stable.txt` to `cilium-cli` and will then change this PR accordingly.

tklauser replied:
Cilium CLI PR add `stable.txt` (and documenting the release process): cilium/cilium-cli#434

tklauser replied:
Updated this PR accordingly.

tklauser committed:
.github/workflows: use latest stable cilium-cli release

Use the stable.txt file in the cilium-cli repo to always fetch the
latest stable release.

Suggested-by: Robin Hahling <robin.hahling@gw-computing.net>
Signed-off-by: Tobias Klauser <tobias@cilium.io>

pchaigno reviewed (approved):
:tada:
